platform/process/tasks/ABProcessTaskServiceInsertRecord.js

```diff
@@ -85,28 +85,40 @@ module.exports = class InsertRecord extends InsertRecordTaskCore {
          tasks.push(
             Promise.resolve()
                .then(() => pullTask())
-               .then((val) =>
-                  this._req.retry(() => this.object.model().create(val))
-               )
+               .then((val) => this.object.model().create(val))
+               // NOTE: .create() returns the fully populated instance already.
+               // .then((record) =>
+               //    this.object.model().findAll({
+               //       where: {
+               //          glue: "and",
+               //          rules: [
+               //             {
+               //                key: this.object.PK(),
+               //                rule: "equals",
+               //                value: record[this.object.PK()],
+               //             },
+               //          ],
+               //       },
+               //       populate: true,
+               //    }, null, req)
+               // )
                .then((result) => {
                   results.push(result);
+                  return Promise.resolve();
                })
          );
       });
 
-      return Promise.all(tasks)
-         .then(() => {
-            this.stateUpdate(instance, {
-               data: results,
-            });
-            this.stateCompleted(instance);
-            return true;
-         })
-         .catch((err) => {
-            this.log(instance, "Error completing Insert Record");
-            this.onError(instance, err);
-            throw err;
-         });
+      return Promise.all(tasks).then(
```

Comment:
OK, so what you just deleted, had Error checking and seems to be functionally the same.  

Any reason to remove that?